cmake/preproject.cmake

```diff
@@ -92,3 +92,9 @@ if("${COMPILE_TARGET}" STREQUAL "snp" OR "${COMPILE_TARGET}" STREQUAL "virtual")
 endif()
 
 set(CMAKE_CXX_STANDARD 20)
+
+if("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
```

Comment:
Let's add a check for USE_LIBCXX, it won't change anything but it will make it clearer that this is only applicable then. Or better even, regroup this to ccf_app.cmake:23, where we set the libc++ flags.

I think it would also be slightly more appropriate to use `add_compile_definitions(_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_FAST)`, but the generated commands are the same of course.